src/test/java/de/uslu/ShopServiceTest.java

```diff
@@ -20,6 +20,35 @@ void myAddProductShouldBeAddInNewShopService() {
         ShopService newShopService = new ShopService(orders, productsRepo);
         newShopService.addProduct(product);
         var result = newShopService.getProduct("1");
-        assertEquals(product, result);
+        assertEquals(product, result.get());
+    }
+
+    @Test
+    void shouldAddProduct() {
+
+        Product product = new Product("Banane", "1");
+        HashMap<String, Product> map = new HashMap<>();
+        assertEquals(map.size(), 0);                    // liste/map ist leer um zu beweisen ob das adden funktioniert
+        ProductRepo productsRepo = new ProductRepo(map);
+
+        ShopService newShopService = new ShopService(new OrderRepo(), productsRepo);
+        newShopService.addProduct(product);                                 //hier object hinzugefügt beweis dass adden klappt.
+        var result = newShopService.getProduct("1");
+        assertEquals(product, result.get());
+    }
+
+    @Test
+    void shouldReturnProductList() {
+        Product product1 = new Product("ads", "1");
+        Product product2 = new Product("da", "2");
+        Product product3 = new Product("asdsa", "3");
+        HashMap<String, Product> productsList = new HashMap<>();
+        ProductRepo productRepo = new ProductRepo(productsList);
+        productRepo.add(product1);
+        productRepo.add(product2);
+        productRepo.add(product3);
```

Comment:
```suggestion
```

Should also work without these adds, since you initialize your Repo with a HashMap of the same 3 Products. With this add statements your only overriding your current (identical) products.